specification/_global/knn_search/_types/Knn.ts

```diff
@@ -18,16 +18,25 @@
  */
 
 import { Field } from '@_types/common'
-import { long, float } from '@_types/Numeric'
-import { QueryVector } from '@_types/Knn'
+import { long, float, double } from '@_types/Numeric'
+import { QueryVector, QueryVectorBuilder } from '@_types/Knn'
+import { QueryContainer } from '@_types/query_dsl/abstractions'
 
 export interface Query {
   /** The name of the vector field to search against */
   field: Field
   /** The query vector */
-  query_vector: QueryVector
+  query_vector?: QueryVector
+  /** The query vector builder. You must provide a query_vector_builder or query_vector, but not both. */
+  query_vector_builder?: QueryVectorBuilder
   /** The final number of nearest neighbors to return as top hits */
   k: long
   /** The number of nearest neighbor candidates to consider per shard */
   num_candidates: long
+  /** Boost value to apply to kNN scores */
+  boost?: float
+  /** Filters for the kNN search query */
+  filter?: QueryContainer | QueryContainer[]
+  /** The minimum similarity for a vector to be considered a match */
+  similarity?: double
```

Comment:
@pstrsr hello, according to https://www.elastic.co/guide/en/elasticsearch/reference/8.10/knn-search-api.html the `_knn_search` API doesn't have `similarity` parameter.

I also want to see `similarity` parameter published: https://github.com/elastic/elasticsearch-specification/pull/2293 . Because of that I suggest to revert changes to this file completely.

Reply:
Thanks, it seems that I confused the APIs. Reverted the changes.